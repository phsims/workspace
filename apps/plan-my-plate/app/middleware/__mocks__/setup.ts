export const getSpoonConfig = () => ({
  baseUrl: 'http://mock_base_url.com',
   imgUrl: 'http://mock_img_url.com',
  headers: {
    'Content-Type': 'application/json',
    'X-API-Key': 'http://my_mock_api_key.com',
  },
  env: 'development',
});
