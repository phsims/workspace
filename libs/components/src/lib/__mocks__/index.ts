export * from './nutrientsData'
export * from './recipesData'