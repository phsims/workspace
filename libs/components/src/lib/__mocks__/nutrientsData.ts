export const mockNutrients = [
  {
    name: 'Calories',
    value: 316.49,
  },
  {
    name: 'Fat',
    value: 12.09,
  },
];
