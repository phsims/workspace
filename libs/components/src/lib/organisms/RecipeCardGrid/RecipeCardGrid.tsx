import { Container, Grid } from '@mui/material';
import RecipeCard, {
  RecipeCardProps,
} from '../../molecules/RecipeCard/RecipeCard';

export interface RecipeCardGridProps {
  recipes: RecipeCardProps[];
}

export function RecipeCardGrid({ recipes }: RecipeCardGridProps) {
  return (
    <Container>
      {recipes && (
        <Grid
          container
          spacing={{ xs: 3, md: 5 }}
          sx={{ alignItems: 'stretch' }}
        >
          {recipes.map((recipe: RecipeCardProps) => (
            <Grid
              item
              lg={4}
              sm={6}
              xs={12}
              key={recipe.id}
              sx={{ display: 'flex' ,  alignItems: 'stretch',}}
            >
              <RecipeCard {...recipe} />
            </Grid>
          ))}
        </Grid>
      )}
    </Container>
  );
}

export default RecipeCardGrid;
