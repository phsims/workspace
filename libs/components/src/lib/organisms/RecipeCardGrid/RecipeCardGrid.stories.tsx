import type { Meta, StoryObj } from '@storybook/react';
import { RecipeCardGrid } from './RecipeCardGrid';
import { mockRecipes } from '../../__mocks__/recipesData';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof RecipeCardGrid> = {
  component: RecipeCardGrid,
  title: '@organisms/RecipeCardGrid',
};
export default meta;
type Story = StoryObj<typeof RecipeCardGrid>;

export const Primary = {
  args: {
    ...mockRecipes,
  },
  parameters: {
    nextjs: {
      appDirectory: true,
    },
  },
};

export const Heading: Story = {
  args: { ...Primary.args },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to RecipeCardGrid!/gi)).toBeInTheDocument();
  },
};
