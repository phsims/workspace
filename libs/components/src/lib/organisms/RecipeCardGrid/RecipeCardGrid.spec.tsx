import { render } from '@testing-library/react';
import mockRouter from 'next-router-mock';

import { mockRecipes } from '../../__mocks__/recipesData';

import RecipeCardGrid from './RecipeCardGrid';


jest.mock('next/navigation', () => require('next-router-mock'));

describe('RecipeCardGrid', () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl('/recipes');
  });
  it('should render successfully', () => {
    const { baseElement } = render(<RecipeCardGrid {...mockRecipes} />);
    expect(baseElement).toBeTruthy();
  });
  it('should match the snapshot', () => {
    const { baseElement } = render(<RecipeCardGrid {...mockRecipes} />);
    expect(baseElement).toMatchSnapshot();
  });
  it('should render the correct number of recipes', () => {
    const { getByTestId } = render(<RecipeCardGrid {...mockRecipes} />);
    mockRecipes.recipes.forEach((recipe) => {
      expect(getByTestId(recipe.id)).toBeInTheDocument();
    });
  });
});
