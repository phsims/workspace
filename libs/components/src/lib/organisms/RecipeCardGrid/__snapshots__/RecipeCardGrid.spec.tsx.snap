// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCardGrid should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-spacing-md-5 css-19wpfn7-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1x80qcd-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1l5bmg7-MuiPaper-root-MuiCard-root"
            data-testid="652825"
          >
            <div
              class="MuiBox-root css-1urz026"
            >
              <img
                alt="My Asian Calamari"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="400px"
                src="/_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F652825-556x370.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              />
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              >
                My Asian Calamari
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              >
                My Asian Calamari requires around 
                <b>
                  45 minutes
                </b>
                 from start to fini...
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1sf5ye3-MuiCardActions-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Go to recipe
                <span
                  class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowForwardIosRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.38 21.01c.49.49 1.28.49 1.77 0l8.31-8.31c.39-.39.39-1.02 0-1.41L9.15 2.98c-.49-.49-1.28-.49-1.77 0s-.49 1.28 0 1.77L14.62 12l-7.25 7.25c-.48.48-.48 1.28.01 1.76"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1x80qcd-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1l5bmg7-MuiPaper-root-MuiCard-root"
            data-testid="632778"
          >
            <div
              class="MuiBox-root css-1urz026"
            >
              <img
                alt="Artisan Farfalle Pasta With Smoked Salmon and Cream Sauce"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="400px"
                src="/_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F632778-556x370.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              />
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              >
                Artisan Farfalle Pasta With Smoked Salmon and Cream Sauce
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              >
                Artisan Farfalle Pasta With Smoked Salmon and Cream Sauce might be jus...
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1sf5ye3-MuiCardActions-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Go to recipe
                <span
                  class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowForwardIosRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.38 21.01c.49.49 1.28.49 1.77 0l8.31-8.31c.39-.39.39-1.02 0-1.41L9.15 2.98c-.49-.49-1.28-.49-1.77 0s-.49 1.28 0 1.77L14.62 12l-7.25 7.25c-.48.48-.48 1.28.01 1.76"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1x80qcd-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1l5bmg7-MuiPaper-root-MuiCard-root"
            data-testid="657537"
          >
            <div
              class="MuiBox-root css-1urz026"
            >
              <img
                alt="Quick and Easy Caprese Salad"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="400px"
                src="/_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.spoonacular.com%2Frecipes%2F657537-556x370.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              />
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              >
                Quick and Easy Caprese Salad
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
              >
                Need a 
                <b>
                  gluten free, lacto ovo vegetarian, and primal main course
                </b>
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1sf5ye3-MuiCardActions-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Go to recipe
                <span
                  class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowForwardIosRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.38 21.01c.49.49 1.28.49 1.77 0l8.31-8.31c.39-.39.39-1.02 0-1.41L9.15 2.98c-.49-.49-1.28-.49-1.77 0s-.49 1.28 0 1.77L14.62 12l-7.25 7.25c-.48.48-.48 1.28.01 1.76"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
